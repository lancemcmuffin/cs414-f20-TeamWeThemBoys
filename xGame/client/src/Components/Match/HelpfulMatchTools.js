import React from "react";
import { Button } from "reactstrap";
import Grid from "@material-ui/core/Grid";

import Rook from "../../data/images/Rook.png";
import Knight from "../../data/images/Knight.png";
import Bishop from "../../data/images/Bishop.png";
import King from "../../data/images/King.png";
import Queen from "../../data/images/Queen.png";
import Pawn from "../../data/images/Pawn.png";

const PawnPiece = {
  Image: <img src={Pawn} className={"chessPiece"} alt={"Pawn"} />,
  blackName: "\u265F",
  whiteName: "\u2659",
};
const RookPiece = {
  Image: <img src={Rook} className={"chessPiece"} alt={"Rook"} />,
  blackName: "\u265C",
  whiteName: "\u2656",
};
const KnightPiece = {
  Image: <img src={Knight} className={"chessPiece"} alt={"Knight"} />,
  blackName: "\u265E",
  whiteName: "\u2658",
};
const BishopPiece = {
  Image: <img src={Bishop} className={"chessPiece"} alt={"Bishop"} />,
  blackName: "\u265D",
  whiteName: "\u2657",
};
const QueenPiece = {
  Image: <img src={Queen} className={"chessPiece"} alt={"Queen"} />,
  blackName: "\u265B",
  whiteName: "\u2655",
};
const KingPiece = {
  Image: <img src={King} className={"chessPiece"} alt={"King"} />,
  blackName: "\u265A",
  whiteName: "\u2654",
};

export function findLocation(location) {
  let number = location.charCodeAt(0) - 97;
  let letter = 8 - parseInt(location[1]);
  return [letter, number];
}

export function showTable(elements) {
  console.log(elements);
  let table = [];
  for (let row = 0; row < elements.length; row++) {
    table.push(
      <Grid item key={"row " + row} style={{ flex: "1" }}>
        {elements[row]}
      </Grid>
    );
  }
  return table;
}

export function getPieceList(color) {
  let pieceList = [];
  if (color === "black") {
    for (let i = 0; i < 8; i++) {
      pieceList.push(PawnPiece.blackName);
    }
    pieceList.push(RookPiece.blackName);
    pieceList.push(RookPiece.blackName);
    pieceList.push(BishopPiece.blackName);
    pieceList.push(BishopPiece.blackName);
    pieceList.push(KnightPiece.blackName);
    pieceList.push(KnightPiece.blackName);
    pieceList.push(KingPiece.blackName);
    pieceList.push(QueenPiece.blackName);
  } else {
    for (let i = 0; i < 8; i++) {
      pieceList.push(PawnPiece.whiteName);
    }
    pieceList.push(RookPiece.whiteName);
    pieceList.push(RookPiece.whiteName);
    pieceList.push(BishopPiece.whiteName);
    pieceList.push(BishopPiece.whiteName);
    pieceList.push(KnightPiece.whiteName);
    pieceList.push(KnightPiece.whiteName);
    pieceList.push(KingPiece.whiteName);
    pieceList.push(QueenPiece.whiteName);
  }
  return pieceList;
}

export function createBoardPiece(pieceName, color, location, clickFunction) {
  color = color.toLowerCase();
  let className = color + "BoardSquare";
  switch (pieceName) {
    case PawnPiece.blackName:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img src={Pawn} className={"chessPiece"} alt={"Pawn"} />
          </Button>
        </Grid>
      );
    case PawnPiece.whiteName:
      return (
        <Grid item key={location} onClick={clickFunction}>
          <Button className={className}>
            <img src={Pawn} className={"white chessPiece"} alt={"Pawn"} />
          </Button>
        </Grid>
      );
    case RookPiece.blackName:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img src={Rook} className={"chessPiece"} alt={"Rook"} />
          </Button>
        </Grid>
      );
    case RookPiece.whiteName:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img src={Rook} className={"white chessPiece"} alt={"Rook"} />
          </Button>
        </Grid>
      );
    case KnightPiece.blackName:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img src={Knight} className={"chessPiece"} alt={"Knight"} />
          </Button>
        </Grid>
      );
    case KnightPiece.whiteName:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img src={Knight} className={"white chessPiece"} alt={"Knight"} />
          </Button>
        </Grid>
      );
    case BishopPiece.blackName:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img src={Bishop} className={"chessPiece"} alt={"Bishop"} />
          </Button>
        </Grid>
      );
    case BishopPiece.whiteName:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img src={Bishop} className={"white chessPiece"} alt={"Bishop"} />
          </Button>
        </Grid>
      );
    case QueenPiece.blackName:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img src={Queen} className={"chessPiece"} alt={"Queen"} />
          </Button>
        </Grid>
      );
    case QueenPiece.whiteName:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img src={Queen} className={"white chessPiece"} alt={"Queen"} />
          </Button>
        </Grid>
      );
    case KingPiece.blackName:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img src={King} className={"chessPiece"} alt={"King"} />
          </Button>
        </Grid>
      );
    case KingPiece.whiteName:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img src={King} className={"white chessPiece"} alt={"King"} />
          </Button>
        </Grid>
      );
    default:
      return (
        <Grid item key={location}>
          <Button className={className} onClick={clickFunction}>
            <img
              src={Pawn}
              className={"transparent chessPiece"}
              alt={"empty square"}
            />
          </Button>
        </Grid>
      );
  }
}
