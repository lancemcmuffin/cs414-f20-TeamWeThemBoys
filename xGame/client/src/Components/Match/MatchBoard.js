import React, { useState } from "react";
import Grid from "@material-ui/core/Grid";
import PropTypes from "prop-types";
import {
  findLocation,
  createBoardPiece,
  showTable,
} from "./HelpfulMatchTools.js";

import "./MatchStyle.css";

const MatchBoard = (props) => {
  const boardState = props.boardState;
  const activeColor = props.activeColor;
  const [chessBoard, setChessBoard] = useState([]);
  const [piecesOnBoard, setPiecesOnBoard] = useState([]);
  const [piecesToUpdate, setPiecesToUpdate] = useState({ add: [], remove: [] });
  const [selected, setSelected] = useState([]);

  const checkForAdditions = () => {
    let newUpdateList = piecesToUpdate;
    for (let x = 0; x < boardState.length; x++) {
      let matchFound = false;
      for (let y = 0; y < piecesOnBoard.length; y++) {
        if (boardState[x] === piecesOnBoard[y]) {
          matchFound = true;
        }
      }
      if (!matchFound) {
        newUpdateList.add.push(boardState[x]);
      }
    }
    setPiecesToUpdate(newUpdateList);
  };

  const checkForRemoval = () => {
    let newUpdateList = piecesToUpdate;
    for (let x = 0; x < piecesOnBoard.length; x++) {
      let matchFound = false;
      for (let y = 0; y < boardState.length; y++) {
        if (boardState[x] === piecesOnBoard[y]) {
          matchFound = true;
        }
      }
      if (!matchFound) {
        newUpdateList.remove.push(boardState[x]);
      }
    }
    setPiecesToUpdate(newUpdateList);
  };

  //change to alter boardState
  const addSelected = (selectedPieceName, selectedLocation) => {
    console.log("clicked");
    setSelected({
    pieceName: selectedPieceName,
    location: selectedLocation,
    });
  };

  const createBoard = () => {
    let alpha = ["a", "b", "c", "d", "e", "f", "g", "h"];
    let board = [];
    for (let row = 0; row < 8; row++) {
      board.push([]);
      for (let column = 0; column < 8; column++) {
        let tempLocation = alpha[7 - row] + (8 - column).toString();
        if (row % 2 === 0) {
          if (column % 2 === 0) {
            board[row].push(
              createBoardPiece(null, "white", tempLocation, () => {
                addSelected(null, tempLocation);
              })
            );
          } else {
            board[row].push(
              createBoardPiece(null, "black", tempLocation, () => {
                addSelected(null, tempLocation);
              })
            );
          }
        } else {
          if (column % 2 === 0) {
            board[row].push(
              createBoardPiece(null, "black", tempLocation, () => {
                addSelected(null, tempLocation);
              })
            );
          } else {
            board[row].push(
              createBoardPiece(null, "white", tempLocation, () => {
                addSelected(null, tempLocation);
              })
            );
          }
        }
      }
    }
    return board;
  };

  const populateBoard = () => {
    let newBoard = chessBoard;
    for (let piece = 0; piece < piecesToUpdate.add.length; piece++) {
      let index = findLocation(piecesToUpdate.add[piece].location);
      let boardColor = "";
      if (
        (index[0] % 2 === 0 && index[1] % 2 === 0) ||
        (index[0] % 2 !== 0 && index[1] % 2 !== 0)
      ) {
        boardColor = "white";
      } else {
        boardColor = "black";
      }
      let tempPieceName = piecesToUpdate.add[piece].pieceName;
      let tempLocation = piecesToUpdate.add[piece].location;
      let newBoardSquare = createBoardPiece(
        tempPieceName,
        boardColor,
        tempLocation,
        () => {
          addSelected(tempPieceName, tempLocation);
        }
      );
      newBoard[index[1]][index[0]] = newBoardSquare;
    }
    for (let piece = 0; piece < piecesToUpdate.remove.length; piece++) {
      let index = findLocation(piecesToUpdate.remove[piece].location);
      let boardColor = "";
      if (
        (index[0] % 2 === 0 && index[1] % 2 === 0) ||
        (index[0] % 2 !== 0 && index[1] % 2 !== 0)
      ) {
        boardColor = "white";
      } else {
        boardColor = "black";
      }
      let tempLocation = piecesToUpdate.remove[piece].location;
      let newBoardSquare = createBoardPiece(
        null,
        boardColor,
        tempLocation,
        () => {
          addSelected(null, tempLocation);
        }
      );
      newBoard[index[1]][index[0]] = newBoardSquare;
    }
    setPiecesToUpdate({ add: [], remove: [] });
    return newBoard;
  };

  //modify to react to boardState change
  if (chessBoard.length === 0) {
    setChessBoard(createBoard());
    checkForAdditions();
    checkForRemoval();
  }
  if (
    (piecesToUpdate.add.length > 0 || piecesToUpdate.remove.length > 0) &&
    chessBoard.length > 0
  ) {
    setChessBoard(populateBoard());
    setPiecesOnBoard(boardState);
  }

  return (
    <div className={"mainBody"}>
      <Grid container className={"gameBoard"} spacing={0}>
        {showTable(chessBoard)}
      </Grid>
    </div>
  );
};

MatchBoard.propTypes = {
  boardState: PropTypes.array,
  activeColor: PropTypes.string,
};
export default MatchBoard;
