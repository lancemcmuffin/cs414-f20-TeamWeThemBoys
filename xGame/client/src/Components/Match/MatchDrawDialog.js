import { Dialog, DialogContent, DialogContentText, DialogTitle } from '@material-ui/core';
import React from "react";
import { Row, Col } from "reactstrap";
import PropTypes from "prop-types";
import { Button, ButtonGroup } from "@material-ui/core";

import * as userService from "../../service/userService";
import * as matchService from "../../service/matchService";


const MatchDrawDialog = (props) => {
    const {
        showDrawDialog,
        setShowDrawDialog,
        activeMatch,
        activeUser
    } = props;

    const acceptDraw = () => {
        matchService.suggestDraw(activeMatch?.id, activeUser?.id);
        setShowDrawDialog(false);
    };

    const declineDraw = () => {
        matchService.denyDraw(activeMatch?.id, activeUser?.id);
        setShowDrawDialog(false);
    };

    return (
        <Dialog open={showDrawDialog} onClose={() => {setShowDrawDialog(false)}} aria-labelledby="form-dialog-title">
            <DialogTitle id="form-dialog-title">Match Paused</DialogTitle>
            <DialogContent>
                <DialogContentText>
                    Your opponent has suggested a draw!
                </DialogContentText>
                <Row className="pb-3">
                  <Col align="center">
                    <ButtonGroup variant="contained">
                        <Button
                            onClick={acceptDraw}
                        >
                            Accept
                        </Button>
                        <Button
                            color="secondary"
                            onClick={declineDraw}
                        >
                            Deny
                        </Button>
                    </ButtonGroup>
                  </Col>
                </Row>
                
            </DialogContent>
        </Dialog>
    );
};

MatchDrawDialog.propTypes = {
  setShowDrawDialog: PropTypes.func,
  activeUser: PropTypes.object,
  showDrawDialog: PropTypes.bool
};

MatchDrawDialog.defaultProps = {
  activeUser: {},
  setShowDrawDialog: () => {},
  showDrawDialog: true
};

export default MatchDrawDialog;
