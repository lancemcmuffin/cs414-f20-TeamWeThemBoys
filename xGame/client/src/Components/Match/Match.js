import React, { useEffect, useState } from "react";
import { Collapse, Row, Col } from "reactstrap";
import PropTypes from "prop-types";
import { Button } from "@material-ui/core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChess } from "@fortawesome/free-solid-svg-icons";

import MatchPlayerStats from "./MatchPlayerStats";
import MatchBoard from "./MatchBoard";

import "./MatchStyle.css";

const Match = (props) => {
  const { isOpen, activeMatch, setActiveMatch } = props;

  let boardStateDefault = [];

  const [boardState, setBoardState] = useState(boardStateDefault);

  useEffect(() => {
    if (
      activeMatch?.chessBoard &&
      activeMatch.chessBoard !== "This is a chessboard"
    ) {
      const newBoard = parseBoard();
      setBoardState(newBoard);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [activeMatch]);

  const parseBoard = () => {
    const newBoardState = [];
    activeMatch.chessBoard.forEach((row) => {
      row.forEach((piece) => {
        if (piece) {
          const newPiece = {
            pieceName: getPieceName(piece.color, piece.type),
            location: piece.position,
          };
          newBoardState.push(newPiece);
        }
      });
    });
    return newBoardState;
  };

  const getPieceName = (color, type) => {
    const isWhite = color === "WHITE";
    switch (type) {
      case "PAWN":
        return isWhite ? "\u2659" : "\u265F";
      case "ROOK":
        return isWhite ? "\u2656" : "\u265C";
      case "BISHOP":
        return isWhite ? "\u2657" : "\u265D";
      case "KNIGHT":
        return isWhite ? "\u2658" : "\u265E";
      case "KING":
        return isWhite ? "\u2654" : "\u265A";
      case "QUEEN":
        return isWhite ? "\u2655" : "\u265B";
      default:
        break;
    }
  };

  const determineActiveColor = () => {
    return activeMatch?.turnCount % 2 === 0 ? "black" : "white";
  };

  return (
    <Collapse isOpen={isOpen}>
      <Row>
        {activeMatch ? (
          <div className={"fullSize"}>
            <MatchPlayerStats
              playerName={activeMatch?.whitePlayerNickname}
              activeColor={determineActiveColor()}
              turnCounter={activeMatch?.turnCount}
              boardState={boardState}
            />
            <MatchBoard
              boardState={boardState}
              activeMatch={activeMatch}
              activeColor={determineActiveColor()}
              setActiveMatch={setActiveMatch}
            />
            <MatchPlayerStats
              playerName={activeMatch?.blackPlayerNickname}
              activeColor={determineActiveColor()}
              turnCounter={activeMatch?.turnCount}
              boardState={boardState}
              opponent={true}
            />
          </div>
        ) : (
          <Col className="pt-3 pb-3">
            <FontAwesomeIcon icon={faChess} size="9x" />
            <div className="h3 pt-3">No match open</div>
          </Col>
        )}
      </Row>
      <Row>
        <div className={"center"}>
          <Button variant="contained" color="secondary">
            Forfeit
          </Button>
        </div>
      </Row>
    </Collapse>
  );
};

Match.propTypes = {
  isOpen: PropTypes.bool,
  setActiveMatch: PropTypes.func,
  activeMatch: PropTypes.shape({
    id: PropTypes.number,
    whitePlayerId: PropTypes.number,
    blackPlayerId: PropTypes.number,
    whitePlayerNickname: PropTypes.string,
    blackPlayerNickname: PropTypes.string,
    turnCount: PropTypes.number,
    chessBoard: PropTypes.array,
  }),
};

Match.defaultProps = {
  isOpen: true,
  activeMatch: {
    id: 0,
    whitePlayerId: 0,
    blackPlayerId: 0,
    whitePlayerNickname: "",
    blackPlayerNickname: "",
    turnCount: 0,
    chessBoard: [],
  },
};

export default Match;
