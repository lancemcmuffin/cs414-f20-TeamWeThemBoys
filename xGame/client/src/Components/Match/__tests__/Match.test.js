import "../../../enzyme.config.js";
import React from "react";
import { shallow } from "enzyme";

import Match from "../Match";

function testInitialState() {
  const app = shallow(<Match isOpen={true}/>);
  expect(app.html).toMatchSnapshot();
}

test("Testing login initial state", testInitialState);
