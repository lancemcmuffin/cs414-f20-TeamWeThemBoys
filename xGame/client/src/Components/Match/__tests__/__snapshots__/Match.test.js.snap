// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing login initial state 1`] = `"<div class=\\"collapse show\\"><div class=\\"fullSize\\"><div class=\\"sidebar\\"><p>White player name and pieces captured</p></div><div class=\\"mainBody\\"><div class=\\"MuiGrid-root gameBoard MuiGrid-container\\"><div class=\\"MuiGrid-root MuiGrid-item\\" style=\\"flex:1\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div></div><div class=\\"MuiGrid-root MuiGrid-item\\" style=\\"flex:1\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div></div><div class=\\"MuiGrid-root MuiGrid-item\\" style=\\"flex:1\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div></div><div class=\\"MuiGrid-root MuiGrid-item\\" style=\\"flex:1\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div></div><div class=\\"MuiGrid-root MuiGrid-item\\" style=\\"flex:1\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div></div><div class=\\"MuiGrid-root MuiGrid-item\\" style=\\"flex:1\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div></div><div class=\\"MuiGrid-root MuiGrid-item\\" style=\\"flex:1\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div></div><div class=\\"MuiGrid-root MuiGrid-item\\" style=\\"flex:1\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"blackBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"whiteBoardSquare btn btn-secondary\\"><img src=\\"Pawn.png\\" class=\\"transparent chessPiece\\" alt=\\"empty square\\"/></button></div></div></div></div><div class=\\"sidebar\\"><p>Black player name and pieces captured</p></div></div></div>"`;
